### ISSUE TITLE
rustdoc does not warn about misuse of html render option attributes

### ISSUE THREAD

**Issue by lolbinarycat** [rust-lang/rust#149187]
# Code

```rust
#![doc(html_favicon_url)]
#![doc(html_logo_url)]
#![doc(html_playground_url)]
#![doc(issue_tracker_base_url)]
#![doc(html_no_source = "asdf")]
```

# Expected Outcome
Rustdoc should emit a separate warning or error for each attribute misuse.

I would recommend a new lint (eg. `rustdoc::bad_render_attr`), which would be introduced as warn by default, then upgraded to deny by default, and eventually made a hard error over an edition.

# Actual Output
All attributes are silently ignored.

# Version
rustdoc 1.93.0-nightly (53732d5e0 2025-11-20)

# Additional Details
Rustdoc also needs to handle cases like `#![doc(html_logo_url = 3)]` (mismatched type instead of missing argument).

**Comment by lolbinarycat** [comment]
Ok, here's another related snippet:

```rust
#[doc = 1]
mod foo {}
```

**Comment by lolbinarycat** [comment]
Eventually this will all be solved by porting these attributes over to the new parsing infrastructure, but in the meantime I'm gonna try to improve `MetaItemInner`'s ABI.  hopefully changing semantics and implementation separately will make for a smoother transition, since then we should be able to have the rewrite's behavior be identical to what we were doing before.
